marlowe-playground-client/src/Halogen/Blockly.purs

```diff
@@ -147,11 +158,14 @@ handleQuery (GetCodeQuery next) = do
   where
   unexpected s = "An unexpected error has occurred, please raise a support issue at https://github.com/input-output-hk/plutus/issues/new: " <> s
 
-handleAction :: forall m slots. MonadEffect m => Action -> DSL slots m Unit
+handleAction :: forall m slots. MonadAff m => Action -> DSL slots m Unit
 handleAction (Inject rootBlockName blockDefinitions) = do
   blocklyState <- liftEffect $ Blockly.createBlocklyInstance rootBlockName (ElementId "blocklyWorkspace") (ElementId "blocklyToolbox")
   let
     _ =
+      -- QUESTION: why are these functions running on ST? Isn't it enough to have functions return Effect X to indicate
+      -- to the FFI that it is not pure?
+      -- I'm not sure if I'm able to add a Halogen subscription from ST.
```

Comment:
this was an idea while ago to avoid Effect in favour of ST. The benefits and morals of it are dubious though so I am happy for you to change to Effect if it's more useable.